keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -262,6 +315,31 @@ type SeccompProfileSet struct {
 
 ## Design Details
 
+
+### Failure and Fallback Strategy
+
+There are different scenarios in which applying seccomp may fail, below are the ones we mapped 
+and their outcome once this KEP is implemented:
+
+| Scenario | API Server Result | Kubelet Result |
+|--------------------------------------------------------------|------------------------------|----------------------------------------------------------------------------------------------|
+| 1) Using custom or `runtime/default` profile when container runtime does not support seccomp. | Pod created | The outcome is container runtime dependent. In this scenario containers may 1) fail to start or 2) run normally without having its policies enforced. |
+| 2) Using custom or `runtime/default` profile that restricts syscalls a container is trying to make. | Pod created | The outcome is workload and seccomp dependent. In this scenario containers may 1) fail to start, 2) misbehave or 3) log violations. |
+| 3) Using custom profile that relies on unsupported or invalid seccomp actions (i.e. `SCPM_ACT_LOG` in versions earlier than 4.14 Linux kernel). | Pod created | Containers fail to start. Retry respecting RestartPolicy and back-off delay. |
+| 4) Using custom profile that does not exist on the node's disk. | Pod created | Containers fail to start. Retry respecting RestartPolicy and back-off delay. |
+| 5) Using a non supported runtime profile (i.e. `runtime/default-audit`). | Pod **not** created. | N/A |
+| 6) Using localhost profile with invalid name (i.e. "/etc/profilename") | Pod **not** created. | N/A |
+
+Scenario 2 is the expected behavior of using seccomp and it is included here for completeness.
+
+Scenario 6 represents the case of failing the existing validation, which is defined at [Pod API](#pod-api).
+
+### Seccomp root path configuration
+
+The existing kubelet (alpha) flag `--seccomp-profile-root` allows for seccomp root path configuration.
+This flag will be marked as deprecated as of v1.19, and will be removed on v1.23.
+The seccomp root path will then be derived from the kubelet root path, which is defined by `--root-dir`.
+
```

Comment:
@derekwaynecarr @liggitt @tallclair this is the new paragraph on '--seccomp-profile-root' being deprecated.

Reply:
nit: note that the default value for the `--seccomp-profile-root` is `<root-dir>/seccomp`. So the proposal is to make the default behavior the only behavior.

Reply:
Added that information to the same section.

Reply:
I am happy with that.